test/libyul/yulOptimizerTests/circularReferencesPruner/trivial.yul

```diff
@@ -0,0 +1,8 @@
+{
+    function f() -> x { x := g() }
+    function g() -> x { x := f() }
+}
+// ====
+// step: circularReferencesPruner
+// ----
+// { }
```

Comment:
Note to self: This optimization step does not remove unreferenced recursive calls. Example

```
function foo() { foo() }
```

Reply:
I should have tested this case before.

Should I change this optimization step to make it remove unreferenced recursive calls?

Reply:
Yes, would be good!

Reply:
@bshastry I have added this test file and ran `soltest` on it, and it's running successfully:
```
{
	function foo() { foo() }
}
// ====
// step: circularReferencesPruner
// ----
// { }
```

Is this the same test that you have tried? Or did I misunderstand your point?

Reply:
Sorry, I'm not sure why I made the earlier comment. The new test case looks good to me!